src/serializer/ResidualHeapSerializer.js

```diff
@@ -695,6 +696,99 @@ export class ResidualHeapSerializer {
     return t.arrayExpression(initProperties);
   }
 
+  _getPropertyValue(object: any, key: string, serialzeValue: boolean) {
+    if (object != null && object.has(key) === true) {
+      let val = object.get(key);
+
+      if (val !== undefined) {
+        let descriptor = val.descriptor;
+
+        if (descriptor !== undefined) {
+          let descriptorValue = descriptor.value;
+
+          if (descriptorValue !== undefined) {
+            return serialzeValue === true ? this.serializeValue(descriptorValue) : descriptorValue;
+          }
+        }
+      }
+    }
+    return null;
+  }
+
+  _serializeValueReactElement(val: ObjectValue): BabelNodeExpression {
+    let objectProps = val.properties;
+
+    // we do this so they don't get emited
+    this.serializedValues.add(val.$Prototype);
+    let $$typeof = this._getPropertyValue(objectProps, "$$typeof", false);
+    if ($$typeof !== null) {
+      this.serializedValues.add(($$typeof: any));
```

Comment:
Again, avoid adding `serializedValues` like this. Also, seeing all the : any casts hurts my eyes...

Reply:
Also, this only captures the case where `$$typeof` is a data property. When it is an accessor prorperty with getters and setters, your `_getPropertyValue` returns `null`, the value gets skipped here, and the invariant connecting the visitor and the serializer will break.

I suggest you handle this in the visitor:
- Don't visit values that won't be needed.
- If `$$typeof` must be a data properties, and it is not reasonable to support accessor properties, call `logError` (or issue the error in a different way that @hermanventer approves).